contracts/validator-utils/src/signature.rs

```diff
@@ -1,23 +1,41 @@
-use crate::{error::Error, search_cells::search_lock_hashes};
+use crate::{cells::utils::search_lock_hashes, error::Error};
 use ckb_std::{ckb_constants::Source, syscalls::load_cell_data};
 use gw_common::H256;
 
-pub fn check_input_account_lock(
-    account_script_hash: H256,
-    expected_message: H256,
+pub enum SignatureType {
+    Transaction,
+    Message(H256),
+}
+
+/// Check l2 account signature cell
+pub fn check_l2_account_signature_cell(
+    script_hash: &H256,
+    sig_type: SignatureType,
 ) -> Result<(), Error> {
-    // check inputs has accout lock cell
-    for index in search_lock_hashes(&account_script_hash.into(), Source::Input) {
-        // load message from input
-        let mut message = [0u8; 32];
-        let loaded_len = load_cell_data(&mut message, 0, index, Source::Input)?;
-        if message.len() != loaded_len {
-            // the message is corrupted, the signature verification script will return failure
-            return Err(Error::InvalidAccountLockCell);
-        }
-        if H256::from(message) == expected_message {
-            return Ok(());
+    // search layer2 account lock cell from inputs
+    for index in search_lock_hashes(&(*script_hash).into(), Source::Input) {
+        match sig_type {
+            SignatureType::Transaction => {
+                // expected data is 0
+                let len = load_cell_data(&mut [], 0, index, Source::Input)?;
```

Comment:
The ```parse_data()``` in ```eth-account-lock``` requires that data must be either 32 or 64, but here we expect data is 0?

Reply:
You are right, this is an error.